libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1370,17 +1376,42 @@ void ExpressionCompiler::appendBitOperatorCode(Token::Value _operator)
 	}
 }
 
-void ExpressionCompiler::appendShiftOperatorCode(Token::Value _operator)
+void ExpressionCompiler::appendShiftOperatorCode(Token::Value _operator, Type const& _leftType, Type const& _rightType)
 {
-	BOOST_THROW_EXCEPTION(InternalCompilerError() << errinfo_comment("Shift operators not yet implemented."));
+	// stack: rvalue lvalue
```

Comment:
Because lvalue and rvalue have special meaning in the context of compilers, but here, they only refer to the left and right side of the shift expression. It is true that in `a <<= b`, the `a` has to be an lvalue, but this is not true for `a << b`.